iotdb/src/main/java/org/apache/iotdb/db/query/executor/GroupByWithOnlyTimeFilterDataSet.java

```diff
@@ -0,0 +1,281 @@
+          break;
+        }
+      }
+
+      //page data
+      long minTime = pageHeader.getMinTimestamp();
+      long maxTime = pageHeader.getMaxTimestamp();
+      //no point in sequence data with a timestamp less than endTime
+      if (minTime >= endTime) {
+        hasCachedSequenceData = true;
+        batchData = sequenceReader.nextBatch();
+        break;
+      }
+
+      if (canUseHeader(minTime, maxTime, unsequenceReader, function)) {
+        //cal using page header
+        function.calculateValueFromPageHeader(pageHeader);
+        sequenceReader.skipPageData();
+      } else {
+        //cal using page data
+        batchData = sequenceReader.nextBatch();
+        function.calculateValueFromPageData(batchData, unsequenceReader, endTime);
+        if (batchData.hasNext()) {
+          hasCachedSequenceData = true;
+          break;
+        }
+      }
+
+    }
+
+    function.calculateValueFromUnsequenceReader(unsequenceReader, endTime);
+    hasCachedSequenceDataList.set(idx, hasCachedSequenceData);
+    batchDataList.set(idx, batchData);
+    return function.getResult().deepCopy();
+  }
+
+  //skip the points with timestamp less than startTime
+  private void skipExessData(int idx, IAggregateReader sequenceReader,
+      IPointReader unsequenceReader)
+      throws IOException {
+    BatchData batchData = batchDataList.get(idx);
+    boolean hasCachedSequenceData = hasCachedSequenceDataList.get(idx);
+
+    //skip the unsequenceReader points with timestamp less than startTime
+    while (unsequenceReader.hasNext() && unsequenceReader.current().getTimestamp() < startTime) {
+      unsequenceReader.next();
+    }
+
+    //skip the cached batch data points with timestamp less than startTime
+    if (hasCachedSequenceData) {
+      while (batchData.hasNext() && batchData.currentTime() < startTime) {
+        batchData.next();
+      }
+    }
+    if (hasCachedSequenceData && !batchData.hasNext()) {
+      hasCachedSequenceData = false;
+    } else {
+      return;
+    }
+
+    //skip the points in sequenceReader data whose timestamp are less than startTime
+    while (sequenceReader.hasNext()) {
+      PageHeader pageHeader = sequenceReader.nextPageHeader();
+      //memory data
+      if (pageHeader == null) {
+        batchData = sequenceReader.nextBatch();
+        hasCachedSequenceData = true;
+        while (batchData.hasNext() && batchData.currentTime() < startTime) {
+          batchData.next();
+        }
+        if (batchData.hasNext()) {
+          break;
+        } else {
+          hasCachedSequenceData = false;
+          continue;
+        }
+      }
+      //timestamps of all points in the page are less than startTime
+      if (pageHeader.getMaxTimestamp() < startTime) {
+        sequenceReader.skipPageData();
+        continue;
+      }
+      //timestamps of all points in the page are greater or equal to startTime, don't need to skip
+      if (pageHeader.getMinTimestamp() >= startTime) {
+        break;
```

Comment:
Can break?
If break, is'nt that the codes below are valid?

Reply:
break will skip the while loop.